qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,75 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, ignore_set_layout=False):
+        """Create a Statevector from a quantum circuit.
+
+        Args:
+            circuit (QuantumCircuit): A quantum circuit
+            ignore_set_layout (bool): When set to ``True``, if the input ``circuit``
+                has a layout set, it will be ignored. Defaults to ``False``.
+
+        Returns:
+            Statevector: The statevector representing the circuit evolution
+
+        Example:
+            Create a statevector from a transpiled circuit:
+
+            .. code-block:: python
+
+                from qiskit import QuantumCircuit, transpile
+                from qiskit.quantum_info import Statevector
+                from qiskit.providers.basic_provider import BasicSimulator
+
+                qc = QuantumCircuit(3)
+                qc.h(0)
+                qc.cx(0, 1)
+                qc.swap(1, 2)
+
+                backend = BasicSimulator()
+                transpiled1 = transpile(qc, backend, optimization_level=0)
+                transpiled2 = transpile(qc, backend, optimization_level=2)
+
+                # Get statevectors accounting for layout changes
+                sv1 = Statevector.from_circuit(transpiled1)
+                sv2 = Statevector.from_circuit(transpiled2)
+
+                # These will be equivalent up to global phase
+                print(sv1.equiv(sv2))  # True
+        """
+
+        from qiskit.synthesis.permutation.permutation_utils import _inverse_pattern
+
+        # Handle layout extraction
+        layout = None
+        if not ignore_set_layout:
+            layout = getattr(circuit, "_layout", None)
```

Comment:
`QuantumCircuit._layout` always exists, so we wouldn't need `getattr`.  But also, we can use the public attribute here, which would be
```python
if not ignore_set_layout:
    layout = circuit.layout
```